src/components/Articles/CompanyProvisions.vue

```diff
@@ -0,0 +1,227 @@
+import { ConfirmDialog } from '@/components/dialogs'
+
+@Component({
+  components: {
+    ConfirmDialog
+  }
+})
+export default class CompanyProvisions extends Mixins(CommonMixin) {
+  private draftProvisionsRemoved: boolean = false
+  private isEditing: boolean = false
+  private haveChanges: boolean = false
+  private originalProvisionsRemovedValue: boolean = false
+  private isInvalid = false
+  private checkBoxColor = '#d3272c' // $app-red
+
+  // Props
+  @Prop({ default: () => { return false } })
+  private provisionsRemoved!: boolean
+
+  // Emitters
+  @Emit('companyProvisionsChanged')
+  private emitCompanyProvisionsChanged (provisionsremoved: boolean): void {}
+
+  @Emit('haveChanges')
+  private emitHaveChanges (haveChanges: boolean): void {}
+
+  // Watchers
+  @Watch('provisionsRemoved', { deep: true, immediate: true })
+  private onProvisionsRemovedPropValueChanged (): void {
+    if (!this.haveChanges) {
+      this.originalProvisionsRemovedValue = this.provisionsRemoved
+      this.draftProvisionsRemoved = this.provisionsRemoved
+      this.emitHaveChanges(this.haveChanges)
+    }
+  }
+
+  @Watch('draftProvisionsRemoved', { deep: true, immediate: true })
+  private onDraftProvisionsRemovedPropValueChanged (): void {
+    this.isInvalid = false
+  }
+
+  private setCompanyProvisionsDone (): void {
+    if (this.draftProvisionsRemoved !== this.provisionsRemoved) {
+      this.isEditing = false
+      if (this.hasDraftProvisionsRemovedChanged) {
+        this.haveChanges = true
+      } else {
+        this.haveChanges = false
+      }
+      this.emitHaveChanges(this.haveChanges)
+      this.emitCompanyProvisionsChanged(this.draftProvisionsRemoved)
+      this.isInvalid = false
+    } else {
+      this.isInvalid = true
+    }
+  }
+
+  private get hasProvisionsRemovedPropsChanged (): boolean {
+    return this.provisionsRemoved !== this.originalProvisionsRemovedValue
+  }
+
+  private get hasDraftProvisionsRemovedChanged (): boolean {
+    return this.draftProvisionsRemoved !== this.originalProvisionsRemovedValue
+  }
+
+  private cancelCompanyProvisionChange () {
+    this.draftProvisionsRemoved = this.provisionsRemoved
+    this.isInvalid = false
+    this.isEditing = false
+  }
+
+  private resetCompanyProvisions () {
+    this.haveChanges = false
+    this.draftProvisionsRemoved = this.originalProvisionsRemovedValue
+    this.emitCompanyProvisionsChanged(this.draftProvisionsRemoved)
+    this.emitHaveChanges(false)
+    this.isEditing = false
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+.action-btns {
+  display: flex;
+  justify-content: flex-end;
+  padding-bottom: 1rem;
+  padding-right: 0.5rem;
+
+  .v-btn + .v-btn {
+    margin-left: 0.5rem;
+  }
+
+  .v-btn {
+    min-width: 6.5rem;
+  }
+
+  .v-btn[disabled] {
+    color: white !important;
+    background-color: #1669bb !important;
+    opacity: 0.2;
+  }
+}
+.instructions-paragraph {
+  margin-bottom: 1.875rem;
```

Comment:
Our convention has been for a lower element to specify the whitespace above it, instead of an upper element to specify the whitespace below it.

Also, try to use the built-in Vuetify classes for margins and padding... it makes it much easier to fix layout right in the template above instead of having to look up custom classes all the time.

https://vuetifyjs.com/en/styles/spacing/

Reply:
- I tried to use those built-in vuetify classes but there's no one with 30px as specified by UX. That's why I added this class. 
- If we need to set the whitespace to the lower element, should I set the bottom margin/padding to 0px for the upper element?

Reply:
Yes, I know that there is no equivalent for 30px... oh well.

Yes, add the whitespace to the lower element and remove any extra whitespace from the upper element. There are a few exceptions... for example, this self-contained date block has internal bottom padding to have a gutter all around, but no bottom margin.

![image](https://user-images.githubusercontent.com/10002889/110176564-5c3eb500-7db8-11eb-9287-fb732ece5e0d.png)

Reply:
Done.